### DISCUSSION TITLE
Expose a way to destroy cached data

### DISCUSSION THREAD

**Discussion by Angelelz** [TanStack/router#250] [Ideas]
First of all, thanks for creating and sharing a great routing solution!

The loader is a great feature, and preloading links is even better!
But I was looking all around to find a way to destroy or delete the cache without success. The use case is caching data for an authenticated user, and being able to delete it on logout. On my testings, using the [react-location-simple-cache](https://github.com/TanStack/react-location/tree/main/packages/react-location-simple-cache) package it is possible to destroy the data using the `removeAll()` method, but not the preloaded data. So if a user logs in, preloads some data by hovering over links, and then logs out, user 2 can login and see cached data.

There is potential data leakage if the preloaded data maxAge is too long and/or users share the same device.

**Comment by nick-kang** [comment]
Does calling `window.location.reload()` work?
